cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst

```diff
@@ -130,3 +129,211 @@ See the Apache™ Twill®
 `API documentation <http://twill.incubator.apache.org/apidocs/org/apache/twill/filesystem/Location.html>`__
 for additional information about the ``Location`` abstraction.
 
+==================
+PartitionedFileSet
+==================
+
+While a FileSet is a convenient abstraction over actual file system interfaces, it still requires
+the application to be aware of file system paths. For example, an application that maintains data
+over time might have a new file for every month. One could come up with a naming convention that encodes
+the month into each file name, and share that convention across all applications that use this file set.
+Yet that can become tedious to manage, especially if the naming convention should ever change—then all
```

Comment:
I have added to the Sphinx build's ``conf.py`` a replacement for en-dashes (``|--|``) and em-dashes (``|---|``) that you can use here if you'd like, to make it clear what this character is:
```
naming convention should ever change |---| then all
```